.gitignore

```diff
@@ -0,0 +1,12 @@
+# ipfs can generate profiling dump files
```

Comment:
Adding a .gitignore file makes sense, thanks.

A couple things:
1) It should be in its own commit (or PR)
2) This looks like it was taken from go-ipfs which means we should both change the comment here and make it a bit more targeted to the types of outputs that tend to show up in this repo